src/effects/VST3/VST3Effect.cpp

```diff
@@ -365,32 +364,32 @@ bool VST3Effect::SaveSettings(
 }
 
 bool VST3Effect::LoadSettings(
-   const CommandParameters & parms, EffectSettings &settings) const
+   const CommandParameters& parms, EffectSettings& settings) const
 {
    using namespace Steinberg;
 
-   if (mComponentHandler == nullptr)
-      return false;
+   VST3EffectSettings ownSettings;
 
    ForEachParameter
    (
-      [&](const ParameterInfo& pi)
+      [&](const ParameterInfo& parameterInfo)
       {
-         const auto id = pi.id;
-         const auto value = pi.defaultNormalizedValue;
-
-         if (mComponentHandler->beginEdit(id) == kResultOk)
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
          {
-            auto cleanup = finally([&] {
-               mComponentHandler->endEdit(id);
-            });
-            mComponentHandler->performEdit(id, value);
+            const auto key = VST3Utils::MakeAutomationParameterKey(parameterInfo);
```

Comment:
When you correct the lookup into `parms`, also remove `ParseAutomationParameterKey` which becomes unused

Reply:
ok